### ISSUE TITLE
Facebook Login - Obtain null Authentication token

### ISSUE THREAD

**Issue by DRPrincess** [facebook/facebook-android-sdk#1132]
### Checklist before submitting a bug report

- [X] I've updated to the latest released version of the SDK
- [X] I've searched for existing [Github issues](https://github.com/facebook/facebook-android-sdk/issues)
- [X] I've looked for existing answers on [Stack Overflow](https://facebook.stackoverflow.com), the [Facebook Developer Community Forum](https://developers.facebook.com/community/) and the [Facebook Developers Group](https://www.facebook.com/groups/fbdevelopers)
- [X] I've read the [Code of Conduct](CODE_OF_CONDUCT.md)
- [X] This issue is not security related and can safely be disclosed publicly on GitHub

### Java version

java 11.0.12

### Android version

Andriod 11

### Android SDK version

8.2.0

### Installation platform & version

    implementation 'com.facebook.android:facebook-android-sdk:latest.release'

### Package

Login

### Goals

have an jwt format token

### Expected results

have an jwt format token

### Actual results

authenticationToken is empty

### Code samples & details

```java
val fBLoginManager = com.facebook.login.LoginManager.getInstance()
            fBLoginManager.registerCallback(mFaceBookCallbackManager, object : FacebookCallback<LoginResult> {

                override fun onCancel() {
                    Log.i(TAG, "facebook: onCancel")
               
                }

                override fun onError(error: FacebookException) {
                    Log.i(TAG, "facebook: error$error")
                   
                }


                override fun onSuccess(result: LoginResult) {
                    Log.i(TAG, "facebook: onSuccess")
                    //not empty
                    val accessToken = result.accessToken
                    //empty
                    val authenticationToken = result.authenticationToken
                   
               }
                    
            })
            fBLoginManager.logInWithReadPermissions(mFragment.fragment, mFaceBookCallbackManager, listOf("email", "public_profile"))
```

**Comment by sonjaBrzak** [comment]
I have the same issue. LoginResult returns the correct accessToken, but the authenticationToken (JWT token) is always null. Seems like a bug in the SDK. Is it possible that this will be fixed soon?

**Comment by DRPrincess** [comment]
@sonjaBrzak  because  no documentation provided for Android ,so I understand facebook android sdk is not support OIDC ,I solve my question by google firrbase ,it can help have a JWT token。

**Comment by sonjaBrzak** [comment]
@DRPrincess you are right, Firebase can help you get the JWT, although the issuer of that token will be google, not facebook, which could be a problem for the validation part of the token on the backend side. Did you solve that issue somehow?

**Comment by ViktorAkselrod** [comment]
I have the same issue.

**Comment by Micpol** [comment]
Do we have any progress or information on this issue? This exact code works perfectly on iOS with iOS facebook SDK, but for android I always get null on the AuthenticationToken.getCurrentAuthenticationToken - same for loginResult.authenticationToken.

This is blocking my team from adding facebook login to the app I'm working on.

**Comment by codefluencer** [comment]
@KylinChang @carolineli  @mingcaozhang

Please fix retrieval of JWT token.

It can't be that the same functionality **is working on iOS SDK**, but not on Android and no documentation about this exists!

Why Android SDK is being ignored?

**Comment by avlaev** [comment]
same problem here. Please fix.

**Comment by DougSig** [comment]
Same issue here. Does Meta have any official workaround guidance? Is it expected that we implement the flow manually (https://developers.facebook.com/docs/facebook-login/guides/advanced/oidc-token)?

**Comment by evelant** [comment]
Looks like this is still true. On iOS we can get a standard token and validate it on our backend but we cannot on android.

**Comment by easonHeyuan** [comment]
The problem still exists, with android sdk, I could not get the idToken。is there any other way to get it ?

**Comment by yarodevuci** [comment]
seems like limited token work only on iOS per their new requirements

**Comment by jesseren** [comment]
Thank you for reporting this issue. The SDK version you're using is significantly outdated. Please upgrade to the latest version of the Facebook Android SDK (currently 18.x) and verify whether this issue still occurs.

If the issue persists on the latest version, please reopen this issue with:
- The exact SDK version you're using
- Your Android API level / OS version
- Steps to reproduce the issue

**Comment by spydon** [comment]
Hi @jesseren, following up from the Supabase side. We're hitting this on the latest SDK, and I think the original report undersold it. Here's a precise reproduction.

**Important distinction from the original report:** the original code called `LoginManager.logIn(permissions)` with no nonce, where a null `authenticationToken` is expected behavior. Our case is different: we explicitly request an OIDC token via `LoginConfiguration` **with a nonce**, and still get a null `authenticationToken` on repeat logins.

**Exact SDK version:** `com.facebook.android:facebook-login:18.1.3` (latest 18.x), consumed via the `flutter_facebook_auth` plugin.

**Android API level / OS version:** reproduced across all Android versions and API levels we tested over the last few years (real devices with the Facebook app installed, so native SSO is used).

**Steps to reproduce:**
1. Configure a Facebook app, install the Facebook app on the device and sign in there (so native SSO is used).
2. Log in with a nonce:
```kotlin
val config = LoginConfiguration(
    listOf("public_profile", "email", "openid"),
    /* nonce = */ "<random-nonce>",
)
LoginManager.getInstance().logIn(activity, config)
```
3. **First login** (app not yet authorized on the account): `LoginResult.getAuthenticationToken()` is non-null, an OIDC id token is returned as expected.
4. `LoginManager.getInstance().logOut()`, then log in again with the same configuration: `getAuthenticationToken()` is now **null**, only `getAccessToken()` is populated. The instant/native token path appears to short-circuit and skip the nonce/OIDC flow.
5. The only way to get the id token again is to remove the app under Facebook -> Settings -> Apps and Websites and re-authorize fully.

**Expected:** with a nonce set on `LoginConfiguration`, `getAuthenticationToken()` should be populated on every successful login, as it is on iOS with `LoginTracking.limited`.

**Why this matters / why we need it:** Supabase's `signInWithIdToken` requires an OIDC id token (JWT). Because Facebook's native Android login only returns one on the first authorization, native Facebook login with Supabase is unusable on repeat logins without falling back to the browser (`webOnly`) flow. iOS Limited Login works fine. This blocks a lot of apps.

Related issues for context:
- supabase/supabase-flutter#1287
- darwin-morocho/flutter-facebook-auth#494 (plugin change forwarding the nonce into `LoginConfiguration`)
- supabase/auth#1522
- supabase/auth#2609 (our server-side workaround exchanging the classic access token instead)

Note: since this issue is closed, would you prefer what we move this into a fresh issue?

---

**Update (traced on-device, SDK 18.1.3):** Adding `openid` to the requested permissions does trigger `GetTokenLoginMethodHandler.getTokenCompleted`'s `openid → tryNextHandler` fallback, but the subsequent `KatanaProxyLoginMethodHandler` (native full-auth) still returns a classic access token with `authenticationToken == null` on an already-authorized grant (reproduced under both `NATIVE_WITH_FALLBACK` and `NATIVE_ONLY`). Only a full re-authorization (removing the app under Facebook → Settings → Apps and Websites) yields the id token. So this isn't just the instant get-token path short-circuiting — no native handler mints an OIDC token for a cached grant.
